### PR TITLE
add button to remove entity in trash can

### PR DESCRIPTION
**Pull request by alavieille**
[OO-FEATURE] Remove permanently a entity in trash can

https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1539
https://github.com/open-orchestra/open-orchestra-base-api-bundle/pull/76
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/189
https://github.com/open-orchestra/open-orchestra-model-interface/pull/172
https://github.com/open-orchestra/open-orchestra-model-bundle/pull/541

### CODE REVIEW

File: ApiBundle/Transformer/TrashItemTransformer.php [hunk 1]
```diff
@@ -37,6 +38,13 @@ public function transform($trashItem)
             ));
         }
 
+        if ($this->authorizationChecker->isGranted(AdministrationPanelStrategy::ROLE_ACCESS_DELETED_TRASHCAN)) {
+            $facade->addLink('_self_delete',  $this->generateRoute(
```
Comment (itkg-ngilain) [inline ApiBundle/Transformer/TrashItemTransformer.php]: Suggeston: when ``_self_delete`` is used on the platform, it means a soft delete. Maybe to mark the difference with the hard delete here, we could talk about ``_self_remove``

File: ApiBundle/Controller/TrashcanController.php [hunk 2]
```diff
@@ -58,12 +58,42 @@ public function restoreAction($trashItemId)
...
+     * @param $trashItemId
+     *
+     * @Config\Route("/{trashItemId}/delete", name="open_orchestra_api_trashcan_delete")
+     * @Config\Method({"DELETE"})
+
+     * @Config\Security("is_granted('ROLE_ACCESS_DELETED_TRASHCAN')")
+     *
+     * @return array|mixed
+     */
+    public function deleteAction($trashItemId)
```
Comment (itkg-ngilain) [inline ApiBundle/Controller/TrashcanController.php]: Maybe more a ``removeAction`` with a ``/remove`` url to mark the difference between soft delete and hard delete.

File: Backoffice/DeleteTrashcanEntity/Strategies/DeleteTrashCanContentStrategy.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+     * @param ContentRepositoryInterface $contentRepository
+     * @param EventDispatcherInterface   $eventDispatcher
+     * @param ObjectManager              $objectManager
+     */
+    public function __construct(
+        ContentRepositoryInterface $contentRepository,
+        EventDispatcherInterface $eventDispatcher,
+        ObjectManager $objectManager
+    )
+    {
```
Comment (itkg-ngilain) [inline Backoffice/DeleteTrashcanEntity/Strategies/DeleteTrashCanContentStrategy.php]: ``){``

File: Backoffice/DeleteTrashcanEntity/Strategies/DeleteTrashCanNodeStrategy.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+     * @param EventDispatcherInterface     $eventDispatcher
+     * @param ObjectManager                $objectManager
+     */
+    public function __construct(
+        NodeRepositoryInterface $nodeRepository,
+        TrashItemRepositoryInterface $trashItemRepository,
+        EventDispatcherInterface $eventDispatcher,
+        ObjectManager $objectManager
+    )
+    {
```
Comment (itkg-ngilain) [inline Backoffice/DeleteTrashcanEntity/Strategies/DeleteTrashCanNodeStrategy.php]: ``){``

File: Backoffice/EventSubscriber/DeleteNodeSubscriber.php [hunk 5]
```diff
@@ -16,7 +17,9 @@ class DeleteNodeSubscriber extends AbstractDeleteSubscriber
     public function addNodeTrashCan(NodeEvent $event)
     {
         $node = $event->getNode();
-        $this->createTrashItem($node, $node->getName());
+        $name = $node->getName() . " (".$node->getLanguage()." - v".$node->getVersion().")";
```
Comment (itkg-ngilain) [inline Backoffice/EventSubscriber/DeleteNodeSubscriber.php]: This ``" - v".$node->getVersion()`` is a little annoying: it works well for a french or english user, but may sound weird for another language

File: BackofficeBundle/Resources/translations/messages.fr.yml [hunk 6]
```diff
@@ -612,5 +612,7 @@ open_orchestra_backoffice:
           deleted_at: Supprimé le
           restore_confirm_txt: Vous voulez vous restaurer cette élément ?
```
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/translations/messages.fr.yml]: cet élément

File: BackofficeBundle/Resources/translations/messages.fr.yml [hunk 7]
```diff
@@ -612,5 +612,7 @@ open_orchestra_backoffice:
...
           restore_confirm_title: Restaurer
+          delete_confirm_txt: Vous voulez vous supprimer cette élément définitivement?
```
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/translations/messages.fr.yml]: cet élément

**Comment by alavieille** [comment]
@itkg-ngilain pr updated

**Comment by itkg-ngilain** [comment]
:+1:

**Pull request merged**